core/services/job_runner.go

```diff
@@ -267,6 +267,14 @@ func executeRun(run *models.JobRun, store *store.Store) error {
 
 	if run.Status.Finished() {
 		run.SetFinishedAt()
+		reward := run.Overrides.Amount
+		if reward != nil {
+			earn := models.NewLinkEarned(run.JobSpecID, run.Overrides.Amount)
```

Comment:
Can we also keep track of the job run id? If we approach this like an accounting book, we could always effectively recalculate the amount earning by walking all runs and peeling it off `run.Overrides.Amount`, and simple return `SELECT SUM(amount) FROM link_earned`